lib/new_relic/agent/method_tracer_helpers.rb

```diff
@@ -109,6 +116,22 @@ def namespace_and_location(object, method_name)
         end
         [name, method.source_location, is_class_method]
       end
+
+      # Rails controllers can be a special case because by default, controllers in Rails
+      # automatically render views with names that correspond to valid routes. This means
+      # that a controller method may not have a corresponding method in the controller class.
+      def controller_without_method?(klass, method_name)
+        defined?(ApplicationController) && klass < ApplicationController && !klass.method_defined?(method_name)
+      end
+
+      def controller_namespace_and_location(klass, method_name, is_class_method, name)
+        return [] unless defined?(Rails)
+
+        path = Rails.root.join("app/controllers/#{klass.name.underscore}.rb")
+        return [] unless File.exist?(path)
+
+        [name, [path.to_s, 1], is_class_method]
```

Comment:
`path` should be a string already and not need the `#to_s` call

Reply:
It looks like out of the 4 method arguments, only `klass` is used. Maybe we should refactor this method to only return either a path or a nil, and then the caller itself can say "if I got a nil, return [] but if I got a path, return a 4 item array using the returned path and the 3 items I already have".

Reply:
When testing, the result of calling `Rails.root` is a [Pathname](https://ruby-doc.org/stdlib-2.6.3/libdoc/pathname/rdoc/Pathname.html). I'm not sure of all the implications of that, but because in other cases we get a String, I thought it would be safer. Can you see the downsides to it being kept as a Pathname object?

I'll work on that refactor!

Reply:
Ah,ok. Check to see if `#join` returns a string instead of another Pathname object. A string won't need a to_s but another Pathname will.

Reply:
Looks like `.join` returns another Pathname

Reply:
Perfect. Sorry for the confusion. The to_s stays.

Reply:
@fallwith thoughts on this refactor https://github.com/newrelic/newrelic-ruby-agent/pull/2061/commits/a9201f18defa664cc36cf80d5152b6609f69f490